src/features/wallet/root/ActivityItem.tsx

```diff
@@ -63,34 +196,38 @@ const ActivityItem = ({
         <Box
           width={50}
           height={50}
-          style={{ backgroundColor: colors[type] }}
+          style={{ backgroundColor: iconBackgroundColor(item.type) }}
           justifyContent="center"
           alignItems="center"
           borderTopLeftRadius={isFirst ? 'm' : undefined}
           borderBottomLeftRadius={isLast ? 'm' : undefined}
         >
-          <Rewards width={26} height={26} />
+          {icon(item.type)}
         </Box>
         <Box flex={1} paddingHorizontal="m">
-          <Text fontSize={14} lineHeight={20} fontWeight="500">
-            {titles[type]}
+          <Text
+            variant="body2Medium"
+            color="black"
+            numberOfLines={1}
+            adjustsFontSizeToFit
+          >
+            {titles(item.type)}
           </Text>
-          <Text color="grayExtraLight" fontSize={14} lineHeight={20}>
-            {type === 'add' && animalHash(amount)}
-            {type !== 'add' && <>{round(amount, 2)} HNT</>}
+          <Text
+            color="grayExtraLight"
+            variant="body2"
+            numberOfLines={1}
+            adjustsFontSizeToFit
+          >
+            {amount}
           </Text>
         </Box>
         <Box paddingHorizontal="m">
-          <Text color="graySteel">
-            {formatDistanceToNow(time, {
-              locale: shortLocale,
-              addSuffix: true,
-            })}
-          </Text>
+          {time && <Text color="graySteel">{time}</Text>}
         </Box>
       </Box>
     </TouchableWithoutFeedback>
   )
 }
 
-export default ActivityItem
+export default memo(ActivityItem)
```

Comment:
Nice, I think we should be using this for most future components?

Reply:
definitely for flatlist items